### PR TITLE
Framework: Move the bug badge out of the way

### PR DESCRIPTION
**Pull request by shaunandrews**
Moving the bug badge to it doesn’t obscure the sidebar. Here's what it looks like on `master`:

screen shot 2015-12-07 at 11 01 58 am

This PR moves the badge so it doesn't overlap the sidebar:

screen shot 2015-12-07 at 11 02 21 am

### CODE REVIEW

**Comment by shaunandrews** [comment]
Need to figure out what to do with the badge on the post screen:

screen shot 2015-12-07 at 11 11 08 am

**Comment by rickybanister** [comment]
What happens on a no-sidebar screen, does it play well there?

**Comment by folletto** [comment]
What about moving it all the way to the right instead, or adding padding to the bottom of the sidebar when we're on a badged site?

Feels really troublesome to have it mid-way through the page.

**Comment by blowery** [comment]
> or adding padding to the bottom of the sidebar when we're on a badged site

might consider doing that regardless of the badge, as the status overlays on some browers (chrome) end up obscuring whatever is at the bottom of the sidebar.

**Comment by gwwar** [comment]
Thanks for looking into this! The only place where i see that there's consistent space available, is up on the masterbar. Maybe move it to the top right and add an opacity so it isn't super distracting.
envbadge

**Comment by shaunandrews** [comment]
I think @gwwar has the right idea here. The Masterbar seems like the most logical place for the bug. Its out of the way, and doesn't interfere with the sidebar or (in the case of WordPress.com) the live chat widget.

**Comment by shaunandrews** [comment]
screen shot 2015-12-07 at 9 43 33 pm

Much better. No it doesn't get in the way of anything.

**Comment by folletto** [comment]
I'm not very keen in having it in the masterbar. Putting something that needs to stay out of the way in the most visible part of our entire IA isn't ideal. 

Even more because it's not just visible to us and developers, but also to any user testing that will be done on Horizon. Can we find a different solution please?

**Comment by shaunandrews** [comment]
> Can we find a different solution please?

I think this is the best solution. Putting it anywhere on the left will interfere with the sidebar. WordPress.com already shows a chat tab in the bottom right, so that's in the way. In the Masterbar, just off to the side of the new post button is the most reliable, but out of the way place. We could adjust the design of the badge so its not so in your face.

Another way to think of the badge is as a nav item for devdocs. Maybe it makes a lot of sense to have that in the Masterbar? Especially if you consider the users on Horizon seeing it.

**Comment by folletto** [comment]
Can't be in the masterbar. Too visible, wrong IA, will interfere with usability / Horizon tests, and it's just not the place for a thing that should be out of the way. 

I'd rather detach from being an overlay and put it at the bottom / footer of the page then, or change the design so it won't take that much space... or would take exactly the same space as the Chrome UI element which we still have to fix regardless of that. Other solutions are available, we just need to explore them.

**Comment by rickybanister** [comment]
Could it be IN the sidebar? We could add another divider at the bottom called 'Dev' and break the bug into multiple links?

Github
Dev Docs

**Comment by melchoyce** [comment]
How is having this in the masterbar any different than putting "Sandboxed" in the masterbar?

**Comment by rickybanister** [comment]
@melchoyce are you advocating for more or for less stuff in the masterbar?

**Comment by melchoyce** [comment]
More — It's a quick & easy way to tell a12s where we're at and what's up with our setups.

**Comment by melchoyce** [comment]
Also, I'm sick of stuff in my site, it always gets in the way. I especially still hate the superadmin menu position.

**Comment by folletto** [comment]
It's different in many ways. Even not counting putting a marginal element
in the top spot, "Sandbox" is visible only in very narrow instances (only
to you, only when you are developing, only on sites that you specify
manually) and not to everyone and users (testing,
OS developers, Horizon) too. Plus it was done before we ever consolidated
the masterbar navigation, so wasn't really something we decided on. It's a
work indicator, not an environment indicator.

I'm not pushing for a solution in particular. Let's just find one that works
instead of rushing something in. I'm sure we can find it. :)

-- 
Davide ‘Folletto’ Casali
AUTOMATTIC

**Comment by shaunandrews** [comment]
Just pushed some changes:

screen shot 2015-12-10 at 3 33 19 pm

Hover over, and it highlights and shows a menu:
 
screen shot 2015-12-10 at 3 33 22 pm

**Comment by rickybanister** [comment]
Just make it draggable so it can snap to any of the four corner of the `main` area :P

**Comment by folletto** [comment]
> Just make it draggable so it can snap to any of the four corner of the main area :P

That... reminded me that I've been recently invited in the Twitter Beta... and they exactly do that. It's a single dot that can be moved around:

screen shot 2015-12-14 at 14 19 38

> Hover over, and it highlights and shows a menu

This is a nice direction. What if we do just the icon as Twitter does, and maybe fit it better in the vertical gap so on widescreen pages won't overlap anything?

File: assets/stylesheets/layout/_main.scss [hunk 1]
```diff
@@ -402,91 +402,89 @@ body.newdash div.wordpress-com-extension-promo {
...
+	opacity: 0.4;
 	position: fixed;
-		bottom: 16px;
-		left: 16px;
-	z-index: 999;
+		top: 47px + 12;
+		right: 12px;
 	backface-visibility: hidden;
+	height: 28px;
+	z-index: 9999;
```
Comment (gwwar) [inline assets/stylesheets/layout/_main.scss]: :) Don't forget to add this to the z-index map after you rebase.

File: assets/stylesheets/layout/_main.scss [hunk 2]
```diff
@@ -402,91 +402,89 @@ body.newdash div.wordpress-com-extension-promo {
 }
 
 .environment-badge {
+	opacity: 0.4;
 	position: fixed;
-		bottom: 16px;
-		left: 16px;
-	z-index: 999;
+		top: 47px + 12;
```
Comment (gwwar) [inline assets/stylesheets/layout/_main.scss]: Maybe use $masterbar-height var here? It reads a bit funny otherwise.

**Comment by gwwar** [comment]
I'm really liking the direction that this is headed. Maybe toggle the environment dropdown when clicking on the badge? Right now you need to click anywhere else to clear it.

**Comment by shaunandrews** [comment]
I just pushed some updates to accommodate the new branch name. Also, after some soul searching, I've moved things back to the Masterbar. Its just in the way anywhere else.

screen shot 2015-12-15 at 4 32 01 pm

screen shot 2015-12-15 at 4 33 44 pm

**Comment by shaunandrews** [comment]
We could do a lot more with this (like making it draggable) if we moved it to a component, rather than part of the jade template. @gwwar suggested we could set a meta tag in the jade template, and use that to display a react component — which would give us a lot more control. Something to look into after this PR lands.

**Comment by apeatling** [comment]
I know we have space in the masterbar and it seems obvious to put it there. However I have to strongly disagree with this direction.

We've just come from 6-7 years of heavily modifying the masterbar/toolbar for developers that took it very far away from what our users see. I don't want us to head down that path again because it masked interaction issues our users experienced since we were blinded by our own modifications.

What other options do we have here? Are there ways we can keep it very minimal until it is interacted with? That way we are limiting how many differences we see compared to our users.

**Comment by gwwar** [comment]
> What other options do we have here? Are there ways we can keep it very minimal until it is interacted with? 

Maybe this would be a good candidate to write as a sidebar plugin? Another option might be to write a few browser extensions that can display this information completely outside of the page.

**Comment by mtias** [comment]
> I know we have space in the masterbar and it seems obvious to put it there. However I have to strongly disagree with this direction.

I have to agree with this. My suggestion would be to work together with the translate button:

image

**Comment by shaunandrews** [comment]
Apparently I'm alone on this — but I find the badge super annoying no matter where we put it.

> that took it very far away from what our users see.

I'm pretty sure users do see this on Horizon.

> Another option might be to write a few browser extensions that can display this information completely outside of the page.

I'd love this, but I have no idea how to go about it.

**Comment by folletto** [comment]
> I'm pretty sure users do see this on Horizon.

They do, and it's why it's sensible to not have it so high in the visual hierarchy. :)

**Comment by rickybanister** [comment]
I like @mtias's solution of matching the translate bug. At least there's the start of a pattern there if we have to add things (rather than bugs and dots popping up all over the place).

What if it shows by default and when you hover it there is a third option after devdocs and github to hide it—making the `?` menu (or something) the only way to bring it back. That way there's no possibility of missing its existence on your first visits, but afterward you can hide it entirely to get the pure experience?

**Comment by gwwar** [comment]
> I'd love this, but I have no idea how to go about it.

If people are still interested, I can prototype this when I get back from vacation.

We could also try something simple like tucking more info into the favicon and adding environment info to the title.

**Comment by rralian** [comment]
@shaunandrews I'm gonna mark this as `in-progress`, as it seems there's enough people with concerns about adding this to the masterbar that we should explore other options.

**Comment by lancewillett** [comment]
Let's re-open this one if we decide to move the badge in the future. "Works for me" is current status.

**Pull request closed without merging**